src/libasr/codegen/asr_to_fortran.cpp

```diff
@@ -824,7 +1179,49 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
 
     // void visit_Nullify(const ASR::Nullify_t &x) {}
 
-    // void visit_Flush(const ASR::Flush_t &x) {}
+    void visit_Flush(const ASR::Flush_t &x) {
+        std::string r = indent;
+        r += "flush";
+        r += "(";
+        bool prefix_comma = false;
+        if (x.m_unit) {
+            visit_expr(*x.m_unit);
+            r += s;
+            prefix_comma = true;
+        } else {
+            throw CodeGenError("flush() function must be called with a file unit number");
+        }
+        if (x.m_err) {
+            if (prefix_comma) {
+                r += ", ";
+            }
+            r += "err=";
+            visit_expr(*x.m_err);
+            r += s;
+            prefix_comma = true;
+        }
+        if (x.m_iomsg) {
+            if (prefix_comma) {
+                r += ", ";
+            }
+            r += "iomsg=";
+            visit_expr(*x.m_err);
```

Comment:
```suggestion
            visit_expr(*x.m_iomsg);
```
Make sure that others visit_expr are correct as well.

Reply:
Ooops...

Thanks!